### DISCUSSION TITLE
Audio parameters for different Vocoder and TTS models

### DISCUSSION THREAD

**Discussion by arif334** [coqui-ai/TTS#1254] [General Q&A]
I've been training different models (from the recipe) with my `Bangla (Bengali)` dataset containing 12 hours of clean speech. So far, I've trained `GlowTTS`, `Hifi-GAN`, `MB-MelGAN`, and `WaveRNN`. They are working fine with each other.

But, then I trained `Tacotron2-DDC` and `Tacotron2-DCA`. They generate a somewhat good voice with the default GL vocoder but do not work with the trained neural vocoders (`Hifi-GAN`, `MB-MelGAN`). Combining Tacotron2 with the neural vocoders generates just noise. 

I've noticed some differences in the `audio` parameters of Tacotron2 and the other models. 

In Tacotron2 config, I've found these:
```
signal_norm=false
log_func=np.log
mel_fmax=8000
spec_gain=1.0
```

While in the other models, these values are as follows:
```
signal_norm=true
log_func=np.log10
mel_fmax=null
spec_gain=20
```

My question is, what should I do to be able to use Tacotron2 with other models? Should I retrain the Tacotron2 (and/or a vocoder) model with matching parameters? Changing those values to the trained model's config files (in either Tacotron2 or vocoder models) generates robotic (but intelligible) sound BTW.

**Comment by erogol** [comment]
Audio parameters should match

**Reply by arif334** [reply]
> Audio parameters should match

Thanks @erogol. I've got that. Now I have to retrain one of the models. Which of the following two options would you suggest?
1. Retrain Tacotron2 with matching parameters of a vocoder.
2. Retrain a vocoder with matching parameters of Tacotron2.

**Reply by erogol** [reply]
Matching vocoder would be easier
